cmd/podman/system/service_abi.go

```diff
@@ -34,6 +35,14 @@ func restService(opts entities.ServiceOptions, flags *pflag.FlagSet, cfg *entiti
 		listener = &l
 	}
 
+	// Close stdin, so shortnames will not prompt
+	devNullfile, err := os.Open(os.DevNull)
+	if err != nil {
+		return err
+	}
+
+	os.Stdin.Close()
+	os.Stdin = devNullfile
```

Comment:
This is a Go pointer assignment that AFAICS does nothing with UNIX FDs, FD 0 remains closed and will be reused.